### ISSUE TITLE
Number filter issues

### ISSUE THREAD

**Issue by colorgap** [ceolter/ag-grid#1525]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Column Number filter equals and not equals doesn't provide matching rows.
In Range, lessthan, greaterthan are not working if user input is 0.

**Expected behavior**
Column Number filter equals and not equals should provide row matching.
In Range, lessthan, greaterthan should work if user input is 0.

**Minimal reproduction of the problem with instructions**
https://www.ag-grid.com/example.php
try number filter on any month column, equals and not equals are not working and in range, lessthan, greaterThan, and other are not working if user input value is 0.

* **ag-Grid version:** 8.2.0
was working in v7.x

* **Browser:**
I tried in chrome but i guess its with all browser

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
ES5

**Comment by seanlandsman** [comment]
Hi

We've identified this issue and have a ticket for it - it'll be fixed in our next release

thanks for reporting this anyway! 

thanks

**Comment by stevey1** [comment]
this is not fixed in 9.0.0, right?

**Comment by ajarratt** [comment]
Hey all, also found this issue. Not fixed as of 9.0.2, as far as I can tell. @stevey1

**Comment by makinggoodsoftware** [comment]
Yup, this is now fixed in our source code, we had to do emergency releases that's why I hasn't come out yet, it will, (this time for real 😉) be available in the next release

**Comment by jenb34** [comment]
Hi, is this fix in 9.1.0? No numeric filters with an equals appear to work?

**Comment by makinggoodsoftware** [comment]
Hi,

The bug reported here which was about filtering number columns when filter input was 0 was broken this should be fixed in 9.1.0.

Is this what is not working for you? Or is it a different issue?

The 0 issue I have doubled checked and seems to be there.

**Comment by jenb34** [comment]
Hi :)

Thanks for getting back to me. So the issue I see is that none of the numeric filters that include equals work. E.g. if I filter on "equals 5" and my column contains a 5 then it will not return. If I filter on "less than or equal to 5" then everything below 5 will be returned but not the 5 itself. And so on for the other filters...

I also see an issue whereby only the visible part of the grid is sorted if the column is numeric - is this a known issue?

Thanks :)

**Comment by jenb34** [comment]
Update: the second issue seems to be that although the column is numeric it is being sorted as a string

eg: 1, 5, 7, 11, 22 would become 1, 11, 22, 5, 7

**Comment by makinggoodsoftware** [comment]
It sounds to me like your underlying data is not numerical, it the column that you are telling to apply a numeric filter to it using strings behind the scenes?

If that is the case you could use a valueGetter to transform the values on the fly, or you could massage the data yourself before setting it into the grid so the column contents are numerical.

The filter 'number' only works with numerical data

**Comment by laps1967** [comment]
I have a similar issue, I have an Oracle database field that is Long value, I created a filter 'number' for one of the field, when I try to get the equal value(s) nothing is showing up, but with greater than it is working.  If I change the field to 'text' the equals then works.  This grid is not a server side grid meaning I don't have enableServerSideFilter, but I do have other grids that are server side grid and I do have enableServerSideFilter and I don't have this issue with my long field using the filter 'numeric'.  I am using ag-grid v11.0.

**Comment by seanareed** [comment]
Is there a standard approach for displaying and filtering currency columns in ag-Grid then?

If I'm not mistaken, I have to format my currency values as strings to ensure the display of the decimal point and two subsequent digits for all values.

But if I display the currencies as strings, ag-Grid's "equals" won't work in the filter, right?

Anybody found a workaround for this?

I'm using v15 at the moment.

**Comment by mahecomsci** [comment]
With ag-grid 20.2.0, greaterThan 0 is not working.

**Comment by egorovas** [comment]
I have a similar issue: "ag-grid-angular": "26.2.0", "ag-grid-community": "26.2.1". Number filter with user input '0' is not working.

**Comment by kiril-matev** [comment]
Hi @egorovas,

Thanks for reaching out.

You've not provided the context necessary to reproduce this issue.

However, in v26.2 there was a similar issue which was logged as:
**AG-6052 [Regression] When using an agNumberColumnFilter to filter for values equal to 0, no values are returned**

This issue was fixed in v27.0. Please upgrade to a new version of AG Grid to resolve this.

Thanks again for contacting us about this.
